tests/rptest/context/cloud_storage.py

```diff
@@ -0,0 +1,175 @@
+import abc
+from enum import IntEnum
+from typing import Optional
+
+from ducktape.tests.test import TestContext
+
+# Injected args
+ARG_CLOUD_STORAGE_TYPE_KEY = "cloud_storage_type"
+
+# Cloud storage generic keys
+GLOBAL_CLOUD_STORAGE_CRED_SOURCE_KEY = "cloud_store_cred_source"
+GLOBAL_CLOUD_PROVIDER = "cloud_provider"
+
+# S3 specific keys
+GLOBAL_S3_ACCESS_KEY = "s3_access_key"
+GLOBAL_S3_SECRET_KEY = "s3_secret_key"
+GLOBAL_S3_REGION_KEY = "s3_region"
+
+# ABS specific keys
+GLOBAL_ABS_STORAGE_ACCOUNT = "abs_storage_account"
+GLOBAL_ABS_SHARED_KEY = "abs_shared_key"
+
+
+class CloudStorageType(IntEnum):
+    # Use (AWS, GCP) S3 compatible API on dedicated nodes, or minio in docker
+    S3 = 1
+    # Use Azure ABS on dedicated nodes, or azurite in docker
+    ABS = 2
+
+
+class Credentials(abc.ABC):
```

Comment:
The plan is to migrate SISettings (aka RedpandaService) to use these as well! Focusing on iceberg for now.